### PR TITLE
Changed data type of account key index to uin32

### PR DESCRIPTION
**Pull request by janezpodhostnik**
Work towards onflow/flow-go#6204

## Description

Changed data type of account key index to uin32.

There are still a few casts in the code, that I did not know how to get rid of.

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/stdlib/account.go [hunk 1]
```diff
@@ -699,7 +699,7 @@ func newAccountKeysGetFunction(
 				var err error
 				var accountKey *AccountKey
 				errors.WrapPanic(func() {
-					accountKey, err = provider.GetAccountKey(address, index)
+					accountKey, err = provider.GetAccountKey(address, uint32(index))
```
Comment (janezpodhostnik) [inline runtime/stdlib/account.go:702]: not sure how to do the cast here safely.

File: runtime/stdlib/account.go [hunk 2]
```diff
@@ -894,7 +894,7 @@ func newAccountKeysRevokeFunction(
 				var err error
 				var accountKey *AccountKey
 				errors.WrapPanic(func() {
-					accountKey, err = handler.RevokeAccountKey(address, index)
+					accountKey, err = handler.RevokeAccountKey(address, uint32(index))
```
Comment (janezpodhostnik) [inline runtime/stdlib/account.go:897]: same here.

**Comment by turbolent** [comment]
I had a look at the referenced issue and it sort of makes sense in flow-go. However, do we want to enforce limits of Flow in Cadence? It would be nice to keep Cadence agnostic of particular chain-limits. Even without the proposed changes, it should be possible to check/reject invalid indices.

**Comment by janezpodhostnik** [comment]
> It would be nice to keep Cadence agnostic of particular chain-limits

does this mean you would prefer cadence still having a `uint64` for the key index? `Uint64` seems like a big data type for key indexes.

**Comment by turbolent** [comment]
Maybe it's good to discuss this a bit more synchronously, maybe as part of the next implementation sync? https://www.notion.so/flowfoundation/Cadence-Implementation-Sync-Notes-Agenda-cf6890eb8a9f4692a8c08611856a5af5?pvs=4#e78f967d7c814ea8b0178e83d39b2e4b. Can have it earlier, don't need to wait for a week. Can also have a separate sync about it

File: runtime/stdlib/account.go (hunk 1 above)
Comment (turbolent) [reply runtime/stdlib/account.go]: Fixed in [63abfd5](https://github.com/onflow/cadence/pull/3465/commits/63abfd5ec7b0d11bf142eead83591bf2e6c83ebb)

File: runtime/stdlib/account.go (hunk 2 above)
Comment (turbolent) [reply runtime/stdlib/account.go]: Fixed in [63abfd5](https://github.com/onflow/cadence/pull/3465/commits/63abfd5ec7b0d11bf142eead83591bf2e6c83ebb)

**Comment by turbolent** [comment]
@janezpodhostnik As discussed, I fixed the Cadence `Int` to Go `uint32` conversion in [63abfd5](https://github.com/onflow/cadence/pull/3465/commits/63abfd5ec7b0d11bf142eead83591bf2e6c83ebb).

@SupunS PTAL 🙏

**Review by turbolent — APPROVED** [review]

**Review by SupunS — APPROVED** [review]

**Pull request merged**